### PR TITLE
Allow domain name to be referenced on an external page in the navigation

### LINKED ISSUE #6939: "http://builtwith.com/yourblog.com" style URLs do not work on the front-end
### Issue Summary

The front end does not respect navigation URLs where the blog URL forms part of the URL such as with http://builtwith.com/yourblog.com 

### Steps to Reproduce

1. In the Navigation section of the admin area, add a new entry `http://builtwith.com/localhost:2368` (change the `localhost:2368` to match your blog URL), then save
2. Open the front-end of your blog, note that the `href` for the newly added navigation link is `http://localhost:2368` (or your blog URL) not the full `http://builtwith.com/x` URL

### PR DESCRIPTION
**Pull request by sakulstra**
Allow domain name to be referenced on an external page in the navigation

closes #6939
- added a test for the specific issue
- modified the url generation to allow the url in an external pages slug

Don't really get why the tests were failing and now aren't any more - didn't change a thing :/

### CODE REVIEW

File: core/server/config/url.js [hunk 1]
```diff
@@ -205,10 +205,12 @@ function urlFor(context, data, absolute) {
             hostname = baseUrl.split('//')[1] + ghostConfig.paths.subdir;
             if (urlPath.indexOf(hostname) > -1
                 && urlPath.indexOf('.' + hostname) === -1
-                && urlPath.indexOf('mailto:') !== 0) {
+                && urlPath.indexOf('mailto:') !== 0
+                && !urlPath.split(hostname)[0].match(/\./)) {
```
Comment (kirrg001) [inline core/server/config/url.js]: with this new condition you don't need to keep `urlPath.indexOf('.' + hostname) === -1`, it's the same result in the case of sub domain.

you can also remove `urlPath.indexOf('mailto:') !== 0` if you change your match regex to `!urlPath.split(hostname)[0].match(/\.|mailto/)`

suggestion:
I think it's not obvious to understand why we don't make the link relative in this case. can you add a comment to describe what happens if we match `.`

**Comment by kirrg001** [comment]
@ErisDS can this be a valid input in the nav?
`some-external-page.com/my-ghost-blog.com`
If yes, what should happen?

File: core/server/config/url.js (hunk 1 above)
Comment (sakulstra) [inline core/server/config/url.js]: Would probably make the code less readable, but you're right `urlPath.indexOf('.' + hostname) === -1` can be removed and 'mailto:' can be combined into one regexp.

I tried to do so `// check if there is another domain in front` but perhaps this is not clear enough? I rephrased it now, but i'm not sure if it's better now :/

**Comment by sakulstra** [comment]
Yay, while rebasing i added a failing test... Should we try to find a solution or just remove the test?
Actually `http://some-external-page.com/stuff-my-ghost-blog.com-around` and `http://some-external-page.com/my-ghost-blog.com` are already edge cases. `http://my-ghost-blog.com/stuff-my-ghost-blog.com-around` is so much over the edge that I think we could easily ignore it ...

**Comment by kirrg001** [comment]
👍  delete the edge case
